realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -4118,6 +4128,72 @@ public void minimumDate_runOnMainThreadThrows() {
         }
     }
 
+    private void fillDictionaryTests(){
+        realm.executeTransaction(transactionRealm -> {
+            AllTypes allTypes1 = realm.createObject(AllTypes.class);
+            allTypes1.getColumnStringDictionary().put("hello world1", "Test1");
+
+            AllTypes allTypes2 = realm.createObject(AllTypes.class);
+            allTypes2.getColumnStringDictionary().put("hello world1", "Test2");
+
+            AllTypes allTypes3 = realm.createObject(AllTypes.class);
+            allTypes3.getColumnStringDictionary().put("hello world2", "Test2");
+        });
+    }
+
+    @Test
+    public void dictionary_containsKey(){
+        fillDictionaryTests();
+        RealmResults<AllTypes> results = realm.where(AllTypes.class).containsKey(AllTypes.FIELD_STRING_DICTIONARY, "hello world1").findAll();
+        assertEquals(2, results.size());
+        assertEquals("Test1", results.get(0).getColumnStringDictionary().get("hello world1"));
+        assertEquals("Test2", results.get(1).getColumnStringDictionary().get("hello world1"));
+    }
+
+    @Test
+    public void dictionary_containsKeyNonLatin(){
+        fillDictionaryTests();
+        RealmResults<AllTypes> results = realm.where(AllTypes.class).containsKey(AllTypes.FIELD_STRING_DICTIONARY, "델타").findAll();
```

Comment:
Hm interesting. I haven't added tests for non-latin string keys and values in the dictionary tests. Maybe I should 🙊

Reply:
Probably a good idea, whitespace and non-lation characters tend to cause issues in funny places.